elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -142,7 +142,10 @@ default Object loadObject(Class<?> entityClass,
                 scope);
         Iterator<Object> it = results == null ? null : results.iterator();
         if (it != null && it.hasNext()) {
-            return it.next();
+            Object obj = it.next();
+            if (!it.hasNext()) {
+              return obj;
```

Comment:
Does this fix an actual issue?

Reply:
Maybe throw an exception instead of returning null.

Reply:
The contract is for `loadObject` to return null if no object is loaded.